libtenzir/builtins/connectors/curl.cpp

```diff
@@ -26,6 +26,7 @@ namespace tenzir::plugins {
 namespace {
 
 struct http_options {
+  bool octet_stream;
```

Comment:
Why does this need its own flag? To set the Content Type? (If so, you typically simply set the header for this.)

Reply:
This is simply an artifact from debugging because we couldn't get the upload to S3 to work from the pre-signed URL. I will split everything except for new option for save and to into a separate PR.